climateconnect/climateconnect
# Only allow viewing and accepting project join requests with correct permissions

## What and Why
Fixes #1197
We had a bug where anybody could  accept or decline join requests from any project.

## Changes in this PR
- Fixed and used the permission class `ReadWriteSensibleProjectDataPermission` for viewing, accepting and declining project join requests
- Refactored `project_slug` --> `url_slug` in some places for standardization and because it was breaking some code
- Added a view and endpoint for checking whether you have requested to join a specific project. Before the frontend would just request all requesters and then check if the current user is in there. Had to make the new endpoint because viewing all requesters isn't possible for normal users anymore with the new permissions
- Added proper error and feedback messages when sending, accepting or declining a join request
- Put `getMembershipRequests` function into a lib file because it was used in many places
- ran `yarn format`, `yarn lint` and `black`

## Testing steps

1. Create a project with one account
2. Switch to a different account and navigate to the project you just created
4. Add `?show_join_requests=true` to the end of the project url. You should see a message telling you only administrators can view open join requests
5. Request to join the project 
6. Log in on the account who owns the project
7. Go to the project with `?show_join_requests=true` attached to the url
8. You should be able to accept and decline join requests as expected
